index.js

```diff
@@ -11,7 +12,15 @@ const resolveFrom = (fromDir, moduleId, silent) => {
 		throw new TypeError(`Expected \`moduleId\` to be of type \`string\`, got \`${typeof moduleId}\``);
 	}
 
-	fromDir = path.resolve(fromDir);
+	try {
+		fromDir = fs.realpathSync(fromDir);
+	} catch (err) {
+		if (err.code !== 'EINVAL') {
```

Comment:
you're right, `EINVAL` was needed only when used with the `resolve-link-target` lib. All should be thrown when using `fs.realpathSync`. Also added tests for silent.

Thanks for the quick review. If anything else, I'll fix tomorrow. morning.